mozetl/maudau/maudau.py

```diff
@@ -0,0 +1,214 @@
+"""
+Calculate Firefox monthly- and daily-active users.
+"""
+import csv
+import datetime as DT
+import logging
+import os
+
+from pyspark.sql import SparkSession
+import boto3
+import botocore
+import click
+from boto3.s3.transfer import S3Transfer
+from moztelemetry.standards import (
+    filter_date_range,
+    count_distinct_clientids
+)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+DEVELOPMENT = False
+
+STORAGE_BUCKET = "net-mozaws-prod-us-west-2-pipeline-analysis"
+STORAGE_SUB_DIR = 'spenrose/maudau' if DEVELOPMENT else 'mreid/maudau'
+MAUDAU_ROLLUP_BASENAME = "engagement_ratio.csv"
+MAUDAU_SNAPSHOT_TEMPLATE = "engagement_ratio.{}.csv"
+DASHBOARD_BUCKET = "net-mozaws-prod-metrics-data"
+DASHBOARD_KEY = "firefox-dashboard/{}".format(MAUDAU_ROLLUP_BASENAME)
+
+today = DT.date.today()
+yesterday = today - DT.timedelta(1)
```

Comment:
"yesterday" is not used; I deleted it. "today" is used as a default overrideable by the calling script except when it is used to record the date a file is written (which in turn is tied to the state of main_summary on S3).  I do not seee a purpose for overriding that; do you?